protecode/util.py

```diff
@@ -42,12 +43,16 @@ def upload_images(
         info('result: {r}'.format(r=result))
         analysis_result = result.result
 
-        vulnerable_components = list(filter(
-            lambda c: c.highest_major_cve_severity() >= cve_threshold, analysis_result.components()
-        ))
+        vulnerable_components = list()
+
+        for component in analysis_result.components():
+            vulnerabilities = filter(lambda v: not v.historical(), component.vulnerabilities())
+            highest_cve = determine_highest_major_cve_severity(vulnerabilities)
+            vulnerable_components.append((component, highest_cve))
 
         if vulnerable_components:
-            highest_cve = max(map(lambda c: c.highest_major_cve_severity(), vulnerable_components))
+            _, highest_cve = max(vulnerable_components, key=lambda pair: pair[1])
```

Comment:
why bother creating two-tuples, when all we need is the cve?
how about
```
determine_highest_major_cve_severity(
  map(
    lambda r: determine_highest_major_cve_severity(r),
    map(lambda c: c.vulnerabilities())
  ),
)```

In other words:
for each component
- determine maximum for one component's vulnerabilities
from all thus determined CVE values, determine the highest CVE